packages/lib/test/treeUtils/sandbox.test.ts

```diff
@@ -492,3 +493,21 @@ test("isSandboxedNode recognizes ref/prev/next to all be sandboxed nodes or not
     r.a = undefined
   })
 })
+
+test("sandbox commit patches are grouped in a single undo item", () => {
+  const b = new B({ value: 0 })
+  const sandboxManager = sandbox(b)
+  const undoManager = undoMiddleware(b)
+
+  expect(undoManager.undoLevels).toBe(0)
+  expect(undoManager.redoLevels).toBe(0)
+
+  sandboxManager.withSandbox(b, (node) => {
+    node.setValue(1)
+    node.setValue(2)
+    return { return: undefined, commit: true }
+  })
+
+  expect(undoManager.undoLevels).toBe(1)
+  expect(undoManager.redoLevels).toBe(0)
```

Comment:
could we add here a `expect(undoManager.undoQueue).toMatchInlineSnapshot(...)` just to make sure the recorded patch is ok?

Reply:
Good idea, done in 83437a9.